presto-docs/src/main/sphinx/admin/cte-materialization.rst

```diff
@@ -0,0 +1,103 @@
+=================================
+CTE Materialization
+=================================
+
+CTEs (Common Table Expressions) are defined as subqueries that appear in a WITH clause provided by the user.
+Their repeated usage within a query can lead in unnecessary redundant computations, excessive data retrieval, and high resource consumption
+
+To address this, Presto supports CTE Materialization. This feature allows intermediate CTEs to be materialized and effectively stored
+and reused within the scope of the same query.
+
+This improves performance and is especially beneficial when the same CTE is utilized multiple times in a query.
+These CTEs will be stored in temporary tables that are bucketed based on the first projection column
+
+To use this feature the connector used by the query must support the creation of temporary tables. Currently, only the :doc:`/connector/hive` connector offers this capability.
+
+How to use CTE Materialization
+--------------
+
+The following configurations and session properties are provided to enable CTE materialization and fine-tune its settings.
+
+``cte-materialization-strategy``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+    * **Type:** ``string``
+    * **Allowed values:** ``ALL``, ``NONE``, ``HEURISTIC`, ``HEURISTIC_COMPLEX_QUERIES_ONLY``
+    * **Default value:** ``NONE``
+
+    Specifies the strategy to use for materializing Common Table Expressions (CTEs) in queries.
+    ``NONE`` indicates that no CTEs will be materialized.
+    ``ALL`` indicates that all CTEs in the query will be materialized.
+    ``HEURISTIC` greedily materializes the earliest parent CTE which is repeated >= cte_heuristic_replication_threshold times
```

Comment:
@jaystarshot : Do you intend to have a blog article or some doc giving more examples (queries and plans) demonstrating the behavior of these config ?

Reply:
Yes that is a good idea, i can add it as a blog or https://github.com/prestodb/presto/wiki